CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
-  # Documentation sources. Only needed for IDEs.
-  prepend(doc_srcs
-    docs/
-    source.dox)
-
-  file(GLOB_RECURSE other_docs docs/*.md)
-  list(APPEND doc_srcs "${other_docs}")
-  set_property(
-    SOURCE ${doc_srcs}
-    APPEND
-    PROPERTY HEADER_FILE_ONLY
-    true)
-    # Doesn't work
-    # $<OR:$<CONFIG:Debug>,$<CONFIG:Release>>)
-endif()
-
-############################################################
-
-add_with_props(rippled_src_all src/ripple/unity/soci.cpp
-  ${soci_extra_includes})
-
-if (NOT is_msvc)
-  set(no_unused_w -Wno-unused-function)
-else()
-  unset(no_unused_w)
+if (assert)
+  foreach (var_
+      CMAKE_C_FLAGS_RELEASE
+      CMAKE_C_FLAGS_MINSIZEREL
+      CMAKE_C_FLAGS_RELWITHDEBINFO
+      CMAKE_CXX_FLAGS_RELEASE
+      CMAKE_CXX_FLAGS_MINSIZEREL
+      CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+    STRING (REGEX REPLACE "[-/]DNDEBUG" "" ${var_} "${${var_}}")
+  endforeach ()
+endif ()
+
+#[===================================================================[
+   rippled compile options/settings via an interface library
+#]===================================================================]
+
+add_library (opts INTERFACE)
+add_library (Ripple::opts ALIAS opts)
+target_compile_definitions (opts
+  INTERFACE
+    BOOST_NO_AUTO_PTR
+    BOOST_ASIO_HAS_STD_ARRAY
+    BOOST_ASIO_DISABLE_HANDLER_TYPE_REQUIREMENTS
+    $<$<BOOL:${boost_show_deprecated}>:
+      BOOST_ASIO_NO_DEPRECATED
+      BOOST_FILESYSTEM_NO_DEPRECATED
+    >
+    $<$<NOT:$<BOOL:${boost_show_deprecated}>>:
+      BOOST_COROUTINES_NO_DEPRECATION_WARNING
+      BOOST_BEAST_ALLOW_DEPRECATED
+      BOOST_FILESYSTEM_DEPRECATED
+    >
+    BEAST_CHECK_MEMORY_LEAKS=$<BOOL:${beast_check_mem_leaks}>
+    RIPPLE_DUMP_LEAKS_ON_EXIT=$<BOOL:${dump_leaks_on_exit}>
+    $<$<BOOL:${beast_no_unit_test_inline}>:BEAST_NO_UNIT_TEST_INLINE=1>
+    $<$<BOOL:${beast_force_debug}>:BEAST_FORCE_DEBUG=1>
+    $<$<BOOL:${beast_disable_autolink}>:BEAST_DONT_AUTOLINK_TO_WIN32_LIBRARIES=1>
+    $<$<BOOL:${single_io_service_thread}>:RIPPLE_SINGLE_IO_SERVICE_THREAD=1>
+    # doesn't currently compile ? :
+    $<$<BOOL:${verify_nodeobject_keys}>:RIPPLE_VERIFY_NODEOBJECT_KEYS=1>)
+target_compile_options (opts
+  INTERFACE
+    $<$<AND:$<BOOL:${is_gcc}>,$<COMPILE_LANGUAGE:CXX>>:-Wsuggest-override>
+    $<$<BOOL:${perf}>:-fno-omit-frame-pointer>
+    $<$<BOOL:${coverage}>:-fprofile-arcs -ftest-coverage>
+    $<$<BOOL:${profile}>:-p -pg>)
+
+target_link_libraries (opts
+  INTERFACE
+    $<$<BOOL:${coverage}>:-fprofile-arcs -ftest-coverage>
+    $<$<BOOL:${profile}>:-p -pg>)
+
+if (jemalloc)
+  find_package (jemalloc REQUIRED)
+  target_compile_definitions (opts INTERFACE PROFILE_JEMALLOC)
+  target_include_directories (opts SYSTEM INTERFACE ${JEMALLOC_INCLUDE_DIRS})
+  target_link_libraries (opts INTERFACE ${JEMALLOC_LIBRARIES})
+  get_filename_component (JEMALLOC_LIB_PATH ${JEMALLOC_LIBRARIES} DIRECTORY)
+  ## TODO see if we can use the BUILD_RPATH target property (is it transitive?)
```

Comment:
Don't know the immediate answer to this, but in general, properties of an `INTERFACE` thing are transitive which is why modern CMake like them ;)

Reply:
I went ahead and set the target property appropriately. Feel free to cherry pick at https://github.com/JoeLoser/rippled/commit/b55220dbbdfbe80c48f3784cc783806aacf742a5